modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/skin/Utils.java

```diff
@@ -213,8 +212,14 @@ public static int computeTruncationIndex(Font font, String text, double width) {
         return index;
     }
 
-    public static String computeClippedText(Font font, String text, double width,
-                                     OverrunStyle type, String ellipsisString) {
+    public static String computeClippedText(
+        Font font,
+        String text,
+        double width,
+        OverrunStyle type,
+        String ellipsisString,
+        AtomicBoolean textTruncated
```

Comment:
I recommend setting `textTruncated` to false as the first statement (alternatively, add a comment that the caller is expected to initialize it to false).

Reply:
setting to false would be incorrect in the case of multi-line text.

I can add a comment, though it's not a public API and there was no comment/javadoc before.

Reply:
Yes, I see. In that case, it's up to you whether to add a simple comment that this should be initialized by the caller (but maybe it doesn't need to be?)